r-package/grf/R/local_linear_forest.R

```diff
@@ -68,48 +78,116 @@
 #'
 #' @export
 ll_regression_forest <- function(X, Y,
-                                 num.trees = 2000,
-                                 clusters = NULL,
-                                 equalize.cluster.weights = FALSE,
-                                 sample.fraction = 0.5,
-                                 mtry = min(ceiling(sqrt(ncol(X)) + 20), ncol(X)),
-                                 min.node.size = 5,
-                                 honesty = TRUE,
-                                 honesty.fraction = 0.5,
-                                 honesty.prune.leaves = TRUE,
-                                 alpha = 0.05,
-                                 imbalance.penalty = 0,
-                                 ci.group.size = 1,
-                                 tune.parameters = "none",
-                                 tune.num.trees = 10,
-                                 tune.num.reps = 100,
-                                 tune.num.draws = 1000,
-                                 num.threads = NULL,
-                                 seed = runif(1, 0, .Machine$integer.max)) {
-
-  forest <- regression_forest(X, Y,
-                              num.trees = num.trees,
-                              sample.weights = NULL,
-                              clusters = clusters,
-                              equalize.cluster.weights = equalize.cluster.weights,
-                              sample.fraction = sample.fraction,
-                              mtry = mtry,
-                              min.node.size = min.node.size,
-                              honesty = honesty,
-                              honesty.fraction = honesty.fraction,
-                              honesty.prune.leaves = honesty.prune.leaves,
-                              alpha = alpha,
-                              imbalance.penalty = imbalance.penalty,
-                              ci.group.size = ci.group.size,
-                              tune.parameters = tune.parameters,
-                              tune.num.trees = tune.num.trees,
-                              tune.num.reps = tune.num.reps,
-                              tune.num.draws = tune.num.draws,
-                              compute.oob.predictions = FALSE,
-                              num.threads = num.threads,
-                              seed = seed)
-  forest[["tuning.output"]] <- NULL
+                                ll.splits = FALSE,
+                                ll.split.weight.penalty = FALSE,
+                                ll.split.lambda = 0.1,
+                                ll.split.variables = NULL,
+                                ll.split.regulate = TRUE,
```

Comment:
Could you maybe only use one parameter, ll.split.cutoff, to get the behavior you want? For example set default ll.split.cutoff = floor(sqrt(nrow(X)) and write in the docstring you disable this behavior (ll.split.regulate) by setting ll.split.cutoff = 0?

Reply:
Good call. Implementing this now. R.e. the earlier comment, I'm setting overall.beta to 0 in this case just to have a non-NULL parameter to pass to the C++ bindings; sound reasonable?

Reply:
Sounds reasonable!

Reply:
Or to be consistent with the rest of grf, `validate_clusters` just passes an empty vector when it is not used `vector(mode = "numeric", length = 0)`

Reply:
Ok I like that better! Thanks, implementing now and I'll push these changes tonight.